core/src/main/java/cz/o2/proxima/repository/ReplicationConsumerNameFactory.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright 2017-2020 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.repository;
+
+import java.io.Serializable;
+
+/** Consumer name generator. */
+public interface ReplicationConsumerNameFactory extends Serializable {
+
+  /**
+   * Initialize for target {@link AttributeFamilyDescriptor}.
+   *
+   * @param familyDescriptor Target AttributeFamilyDescriptor
+   */
+  void setup(AttributeFamilyDescriptor familyDescriptor);
```

Comment:
It might be better readable and extensible if this `setup` method would accept the configuration `Map` and read the sufix/prefixes itself.

Reply:
cfg can be accessible from `familyDescriptor.getCfg()` but changed a little bit.